### PR TITLE
[mlpack] Fix supports

### LINKED ISSUE #21310: mlpack build failure
**Host Environment**
- OS: Windows
- Compiler: VS2019

`.\vcpkg install mlpack --triplet=x64-windows`

**Failure logs** 
-Error: `mlpack[core] is only supported on 'uwp'`

The install works with vcpkg-2021.05.12 but now there is the above error with the main branch.

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #21310

Update `supports: uwp `as `supports: !uwp`. This is caused by mistake via PR https://github.com/microsoft/vcpkg/pull/17983.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
That's my fault, sorry.

**Comment by JackBoosY** [comment]
```
CMake Error at /mnt/vcpkg-ci/installed/x64-linux/share/Armadillo/ArmadilloConfig.cmake:12 (include):
  include could not find requested file:

    /share/Armadillo/ArmadilloLibraryDepends.cmake
Call Stack (most recent call first):
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:788 (_find_package)
  CMakeLists.txt:294 (find_package)
```

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/20439.

**Comment by JackBoosY** [comment]
I will take over this PR.

File: ports/mlpack/portfile.cmake [hunk 1]
```diff
@@ -11,26 +11,25 @@ vcpkg_from_github(
...
         tools   BUILD_CLI_EXECUTABLES
         openmp  USE_OPENMP
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
```
Comment (JackBoosY) [inline ports/mlpack/portfile.cmake:33]:
```suggestion
    DISABLE_PARALLEL_CONFIGURE
    OPTIONS
```

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Ouch, thanks for the fix!

**Pull request merged**